diff --git a/base/allocator/partition_allocator/page_allocator.h b/base/allocator/partition_allocator/page_allocator.h
index f57beb7..3c6f053 100644
--- a/base/allocator/partition_allocator/page_allocator.h
+++ b/base/allocator/partition_allocator/page_allocator.h
@@ -18,7 +18,7 @@ namespace base {
 #if defined(OS_WIN)
 static const size_t kPageAllocationGranularityShift = 16;  // 64KB
 #else
-static const size_t kPageAllocationGranularityShift = 12;  // 4KB
+static const size_t kPageAllocationGranularityShift = 14;  // 16KB
 #endif
 static const size_t kPageAllocationGranularity =
     1 << kPageAllocationGranularityShift;
@@ -29,7 +29,7 @@ static const size_t kPageAllocationGranularityBaseMask =
 
 // All Blink-supported systems have 4096 sized system pages and can handle
 // permissions and commit / decommit at this granularity.
-static const size_t kSystemPageSize = 4096;
+static const size_t kSystemPageSize = 16384;
 static const size_t kSystemPageOffsetMask = kSystemPageSize - 1;
 static const size_t kSystemPageBaseMask = ~kSystemPageOffsetMask;
 
diff --git a/base/allocator/partition_allocator/partition_alloc.h b/base/allocator/partition_allocator/partition_alloc.h
index 15a501a..ca98540 100644
--- a/base/allocator/partition_allocator/partition_alloc.h
+++ b/base/allocator/partition_allocator/partition_alloc.h
@@ -93,7 +93,7 @@ static const size_t kBucketShift = (kAllocationGranularity == 8) ? 3 : 2;
 // system page of the span. For our current max slot span size of 64k and other
 // constant values, we pack _all_ PartitionAllocGeneric() sizes perfectly up
 // against the end of a system page.
-static const size_t kPartitionPageShift = 14;  // 16KB
+static const size_t kPartitionPageShift = 16;  // 16KB
 static const size_t kPartitionPageSize = 1 << kPartitionPageShift;
 static const size_t kPartitionPageOffsetMask = kPartitionPageSize - 1;
 static const size_t kPartitionPageBaseMask = ~kPartitionPageOffsetMask;
