diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index 7fed8b3..4d99eac 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -56,9 +56,13 @@ if (is_linux && target_sysroot_dir != "") {
     sysroot = "//build/linux/ubuntu_precise_amd64-sysroot"
   } else {
     if (current_cpu == "x64") {
-      sysroot = "//build/linux/debian_wheezy_amd64-sysroot"
+      # sysroot = "//build/linux/debian_wheezy_amd64-sysroot"
+      use_sysroot = false
+      sysroot = ""
     } else if (current_cpu == "x86") {
-      sysroot = "//build/linux/debian_wheezy_i386-sysroot"
+      # sysroot = "//build/linux/debian_wheezy_i386-sysroot"
+      use_sysroot = false
+      sysroot = ""
     } else if (current_cpu == "mipsel") {
       sysroot = "//build/linux/debian_wheezy_mips-sysroot"
     } else if (current_cpu == "arm") {
@@ -76,7 +80,7 @@ if (is_linux && target_sysroot_dir != "") {
     # gcc 4.9. Clang is able to detect and work with the 4.6 headers while
     # gcc is not. This check can be removed if we ever update to a more modern
     # sysroot.
-    assert(is_clang, "sysroot images require clang (try use_sysroot=false)")
+#    assert(is_clang, "sysroot images require clang (try use_sysroot=false)")
 
     _script_arch = current_cpu
     if (_script_arch == "x86") {
